From d795187f1275f7b51fc0bcce5d1a903eb49b9cf6 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Fri, 15 Dec 2023 11:31:46 +0100
Subject: [PATCH 0200/1695] dts: stm32: secure pins for secure peripherals on
 stm32mp135f-dk

These pins are used by RCC MCO and the I2C4. As these peripherals are
used in OP-TEE, secure them.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 51ecbb3213882e4c16a532fbdc1897e774955eb8 dts: stm32: secure pins for secure peripherals on stm32mp135f-dk]
Change-Id: I0bb6d7cb8fded5a7de6a73b4c7f1c8a89156ee67
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
---
 core/arch/arm/dts/stm32mp135f-dk.dts | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp135f-dk.dts b/core/arch/arm/dts/stm32mp135f-dk.dts
index 48c29512f..e7cc5ed82 100644
--- a/core/arch/arm/dts/stm32mp135f-dk.dts
+++ b/core/arch/arm/dts/stm32mp135f-dk.dts
@@ -8,6 +8,7 @@
 
 #include <dt-bindings/clock/stm32mp13-clksrc.h>
 #include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/gpio/stm32mp_gpio.h>
 #include <dt-bindings/regulator/st,stm32mp13-regulator.h>
 #include "stm32mp135.dtsi"
 #include "stm32mp13xf.dtsi"
@@ -82,6 +83,18 @@
 	};
 };
 
+&gpiob {
+	st,protreg = < (TZPROT(9)) >;
+};
+
+&gpiod {
+	st,protreg = < (TZPROT(7)) >;
+};
+
+&gpioe {
+	st,protreg = < (TZPROT(15)) >;
+};
+
 &i2c4 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&i2c4_pins_a>;
-- 
2.47.2

